front/src/Translator/Translator.ts

```diff
@@ -0,0 +1,271 @@
+                currentLanguage = localStorageLanguageObject;
+            }
+        }
+
+        if (!currentLanguage && navigatorLanguage) {
+            const navigatorLanguageObject =
+                navigator.language.length === 2
+                    ? Translator.getLanguageWithoutCountry(navigatorLanguage)
+                    : Translator.getLanguageByString(navigatorLanguage);
+            if (navigatorLanguageObject) {
+                currentLanguage = navigatorLanguageObject;
+            }
+        }
+
+        if (!currentLanguage || currentLanguage === Translator.fallbackLanguage) {
+            return;
+        }
+
+        this.currentLanguage = currentLanguage;
+    }
+
+    /**
+     * Get value on object by property path
+     * @param {string} key Translation key
+     * @param {LanguageObject} object Language object
+     * @returns {string|undefined} Found translation by path
+     */
+    private getObjectValueByPath(key: string, object: LanguageObject): string | undefined {
+        const paths = key.split(".");
+        let currentValue: LanguageObject | string = object;
+
+        for (const path of paths) {
+            if (typeof currentValue === "string" || currentValue[path] === undefined) {
+                return undefined;
+            }
+
+            currentValue = currentValue[path];
+        }
+
+        if (typeof currentValue !== "string") {
+            return undefined;
+        }
+
+        return currentValue;
+    }
+
+    /**
+     * Replace {{ }} tags on a string by the params values
+     * @param {string} string Translation string
+     * @param {{ [key: string]: string | number }} params Tags to replace by value
+     * @returns {string} Formatted string
+     */
+    private formatStringWithParams(string: string, params: TranslationParams): string {
+        let formattedString = string;
+
+        for (const param in params) {
+            const regex = `/{{\\s*\\${param}\\s*}}/g`;
+            formattedString = formattedString.replace(new RegExp(regex), params[param].toString());
+        }
+
+        return formattedString;
+    }
+
+    /**
+     * Get translation by a key and formatted with params by {{ }} tag
+     * @param {string} key Translation key
+     * @param {TranslationParams} params Tags to replace by value
+     * @returns {string} Translation formatted
+     */
+    public trans(key: string, params?: TranslationParams): string {
+        const currentLanguageValue = this.getObjectValueByPath(key, this.currentLanguageObject);
+
+        if (currentLanguageValue) {
+            return params ? this.formatStringWithParams(currentLanguageValue, params) : currentLanguageValue;
+        }
+
+        const fallbackLanguageValue = this.getObjectValueByPath(key, Translator.fallbackLanguageObject);
+
+        if (fallbackLanguageValue) {
+            return params ? this.formatStringWithParams(fallbackLanguageValue, params) : fallbackLanguageValue;
+        }
+
+        return key;
```

Comment:
As the system is designed now, the key will never be something "readable".
Shouldn't we throw an error here (or at least output a warning in the console?)